tests/yaml_exercise_tests.py

```diff
@@ -0,0 +1,216 @@
+import unittest
+from unittest import mock
+import yaml
+from src.yaml_exercise import merge_yamls
+
+
+class YamlExerciseTest(unittest.TestCase):
+    @mock.patch('src.yaml_exercise.read_data_from_main_yaml')
+    def test_should_add_dict_with_existing_keys_and_few_more_new_keys_as_dict_to_main_yaml(self, mock_data_from_main_yaml):
+        configuration_to_add = """
+         country: 'USA'
+         city: 'New-York'
+         street: 'street name'
+         """
+        file_name = 'main_yaml.yaml'
+        mock_data_from_main_yaml.return_value = {
+            'name': 'inbar',
+            'address': {'country': 'Israel',
+                        'city': 'Hod-Hashron'},
+            'job': ['programmer', 'student']
+        }
+        expected = {
+            'name': 'inbar',
+            'address': {'country': 'Israel',
+                        'city': 'Hod-Hashron'},
+            'job': ['programmer', 'student'],
+            'country': 'USA',
+            'city': 'New-York',
+            'street': 'street name'
+        }
+
+        merge_yamls(file_name, configuration_to_add)
+        actual = yaml.safe_load(open(file_name).read())
+
+        self.assertEqual(expected, actual)
+
+    @mock.patch('src.yaml_exercise.read_data_from_main_yaml')
+    def test_should_add_dict_with_existing_keys_and_new_values_as_list_instead_original_dict_main_yaml(self, mock_data_from_main_yaml):
+        configuration_to_add = """
+         country: 'USA'
+         city: 'New-York'
+         """
+        file_name = 'main_yaml.yaml'
+        mock_data_from_main_yaml.return_value = {
+            'name': 'inbar',
+            'address': {'country': 'Israel',
+                        'city': 'Hod-Hashron'},
+            'job': ['programmer', 'student']
+        }
+        expected = {
+            'name': 'inbar',
+            'address': [{'country': 'Israel',
+                        'city': 'Hod-Hashron'},
+                        {'country': 'USA',
+                         'city': 'New-York'}],
+            'job': ['programmer', 'student']
+        }
+
+        merge_yamls(file_name, configuration_to_add)
```

Comment:
How did you know where the user wanna add it? How did you guess the hierarchy? 
By the way with no hierarchy from the user I would expect to get:

            'name': 'inbar',
            'address': {'country': 'Israel',
                        'city': 'Hod-Hashron'},
            'job': ['programmer', 'student'],
            'country': 'USA',
            'city': 'New-York'